core/lib/server/stages.ts

```diff
@@ -165,3 +165,15 @@ export const getStages = ({ communityId, stageId, userId }: CommunityStageProps)
 };
 
 export type CommunityStage = AutoReturnType<typeof getStages>["executeTakeFirstOrThrow"];
+
+export const movePub = (pubId: PubsId, stageId: StagesId, trx = db) => {
+	return autoRevalidate(
```

Comment:
I can't quite remember the context around this, but just want to make sure it covers what we talked about [here](https://knowledgefutures.slack.com/archives/CMAQM0BNV/p1723148283839199) a long time ago. I think using the `.with` didn't delete properly, hence the two calls of 

```
await autoRevalidate(
  trx.deleteFrom("PubsInStages").where("PubsInStages.pubId", "=", pubId)
).execute();
await autoRevalidate(
  trx.insertInto("PubsInStages").values({ pubId, stageId })
).execute();
```
in updatePub?

Reply:
I think the problem wasn't that the delete doesn't work, but that it doesn't necessarily happen before the insert. And I noticed that happening specifically when you update a pub, which attempts to move the pub even if the stage hasn't changed. So in that case, (`pubid`,`stageid`) is the same one being deleted and added and we get an error.  I added the `on conflict do nothing` clause below to handle that, and I haven't noticed any issues with the query in other places! It does successfully remove the pub from the stage it's in, and add a new entry for the stage it's moving to.

Reply:
I don't think we actually did what that thread says either, since the [pub moved trigger](https://github.com/pubpub/v7/blob/cb0a574d96cb5aeaf11c6970f08d2f3511ed153c/core/prisma/migrations/20240411012752_create_trigger_on_pub_stage_join_table/migration.sql#L21-L22) wasn't updated. I don't really want to do it in this PR, but I'm happy to make that change in a followup PR, and at the same time get rid of the `PubsInStages` table and replace it with a `pubs.stage` column.

Reply:
very much pro adding a `pubs.stageId` column instead!